packages/shared/src/cx.ts

```diff
@@ -1,5 +1,15 @@
-export function cx(
-  ...classNames: Array<string | number | boolean | undefined | null>
-) {
-  return classNames.filter(Boolean).join(' ') || undefined;
+type classValue = string | undefined | boolean | null | number;
+
+export function cx(...cssClasses: Array<classValue | classValue[]>) {
+  return (
+    cssClasses
+      .reduce<classValue[]>((acc, className) => {
+        if (Array.isArray(className)) {
+          return acc.concat(className);
+        }
+        return acc.concat([className]);
+      }, [])
```

Comment:
I think `classNames.flat()` would achieve the same result.

Reply:
[Doesn't work on IE](https://caniuse.com/array-flat), is it still a requirement ? 🤔

Reply:
Ah that's true!

Reply:
Yep, it would be better, but I don't want to introduce that in an InstantSearch dependency now